contracts/test/ClaimTest.t.sol

```diff
@@ -578,4 +621,411 @@ contract ClaimTest is AddressRegistry, StateManipulations, TestUtils {
+
+        pool.triggerDefault(address(loanV2), address(debtLockerFactory));
+
+        try pool.claim(address(loanV2), address(debtLockerFactory)) { 
+            assertTrue(false, "Claim before liquidation is done"); 
+        } catch Error(string memory reason) {
+            assertEq(reason, "DL:HCOR:LIQ_NOT_FINISHED");
+        }
+
+        _liquidateCollateral();
+
+        // Malicious entity will send 1 WEI work of collateral to Liquidator
+        DebtLocker debtLocker =  DebtLocker(pool.debtLockers(address(loanV2), address(debtLockerFactory)));
+
+        address liquidator = debtLocker.liquidator();
+        
+        erc20_mint(WBTC, 0, liquidator, 1);
+
+        // Claiming is locked again
+        try pool.claim(address(loanV2), address(debtLockerFactory)) { 
+            assertTrue(false, "Claim before liquidation is done"); 
+        } catch Error(string memory reason) {
+            assertEq(reason, "DL:HCOR:LIQ_NOT_FINISHED");
+        }
+
+        debtLocker.stopLiquidation();
+
+        uint256[7] memory details = pool.claim(address(loanV2), address(debtLockerFactory));
+
+        assertEq(usdc.balanceOf(address(loanV2)), 0);
+
+        assertEq(details[0], 280_135_620000);
+        assertEq(details[1], 0);
+        assertEq(details[2], 0);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 280_135_620000);
+        assertEq(details[6], 719_864_380000);
+
+        assertEq(pool.principalOut(), pool_principalOut - 719_864_380000 - 280_135_620000); // Principal out reduced by full amount
+    }
+
+    function test_claim_liquidatonStop() external {
+        _createLoan(INTEREST_ONLY, UNDER_COLLATERALIZED);
+        _fundLoan();
+        _drawdownLoan();
+
+         // Put in default
+        hevm.warp(block.timestamp + loanV2.nextPaymentDueDate() + loanV2.gracePeriod() + 1);
+
+        pool.triggerDefault(address(loanV2), address(debtLockerFactory));
+
+        try pool.claim(address(loanV2), address(debtLockerFactory)) { 
+            assertTrue(false, "Claim before liquidation is done"); 
+        } catch Error(string memory reason) {
+            assertEq(reason, "DL:HCOR:LIQ_NOT_FINISHED");
+        }
+
+        // Instead of liquidating, the Pool Delegate will prematurely stop the liquidation and take a loss
+        DebtLocker debtLocker =  DebtLocker(pool.debtLockers(address(loanV2), address(debtLockerFactory)));
+
+        debtLocker.stopLiquidation();
+
+        // Now claiming is possible
+        uint256[7] memory details = pool.claim(address(loanV2), address(debtLockerFactory));
+
+        assertEq(usdc.balanceOf(address(loanV2)), 0);
+
+        assertEq(details[0], 0);
+        assertEq(details[1], 0);
+        assertEq(details[2], 0);
+        assertEq(details[3], 0);
+        assertEq(details[4], 0);
+        assertEq(details[5], 0);
+        assertEq(details[6], 1_000_000_000000);   
+
+        assertEq(pool.principalOut(), pool_principalOut - 1_000_000_000000);     
```

Comment:
Still need to demo pullFunds after a bad stopLiquidation

Reply:
It's on the test case below it!